### ISSUE TITLE
tab of tabs / search for open tabs

### ISSUE THREAD

**Issue by bel2125** [notepad-plus-plus/notepad-plus-plus#8640]
### Description of the Issue

This is an idea for an additional feature, that might be helpful for some users (that follow a similar workflow as I do).

I tend to have work on a lot of projects at the same time (modify them, read for support cases, copy snippets between them, ...), so I often have 100+ tabs (files) open. Often files in different projects have the same name (like "index.html", "config.ini", "README.txt", ... similar yet individually customized projects).

### Steps to Reproduce the Issue

Open ~100 random text files (best: Take a folder/source tree with 10 files and create 10 copies).

There will be one tab bar with ~10-15 tabs and an some small [<][>] buttons to scroll till infinity ;-)
Alternatively there is a "Windows..." list, but unfortunately there is no way to search for a window (file) name.

### Enhancement idea

Would it be possible to group tabs. Like in a menu bar, is grouped in main menus and item (File, Edit, Search ..., then File-New, File-Open, ...). The user would be responsible for defining what project/category a file belongs to. Then I can open my desired tab in two steps - first choose the project, then the filename.

Or a [...] button in the tab bar (next to the [<] and [>], where you could type some letters to filter opened file names - and some list of filename and folder to pick from). Probably some search function in the "Windows..." like this might work as well. (This would avoid the need for an artificial "category").

**Comment by sasumner** [comment]
> where you could type some letters to filter opened file names - and some list of filename and folder to pick from

Maybe give the *NavigateTo* plugin a look.

**Comment by sasumner** [comment]
Also, there is already the "session" concept, the "project" concept, the "folder as workspace" concept to "group" work.

@bel2125 Some feedback from you on how you've tried _NavigateTo_ and the things mentioned in the previous sentence -- and how they did or didn't suit -- will keep your issue from being closed.

**Comment by bel2125** [comment]
*NavigateTo* is not *precisely* what I was looking for, although it definitely improves navigation.
I was more thinking of a navigation in "projects" containing files, like a navigation tree.
However, from looking at "NavigateTo", learned navigation can be done in plugins - probably this would be an improvement that can be done in a plugin.

> Also, there is already the "session" concept, the "project" concept, the "folder as workspace" concept to "group" work.

I did not know about this - thanks for this information.
I have to test this, maybe this feature is already there, I just did not find it up to now.
